src/signable/oracle-price.ts

```diff
@@ -1,53 +1,79 @@
+import Big, { RoundingMode } from 'big.js';
 import BN from 'bn.js';
 
-import { isoTimestampToEpochSeconds } from '../helpers';
+import {
+  getSignedAssetId,
+  getSignedAssetName,
+  isoTimestampToEpochSeconds,
+} from '../helpers';
 import {
   decToBn,
   hexToBn,
   intToBn,
-  utf8ToBn,
 } from '../lib/util';
 import {
   OraclePriceWithAssetName,
-  OraclePriceWithAssetId,
+  OraclePriceWithMarket,
+  StarkwareOraclePrice,
 } from '../types';
-import { ORACLE_PRICE_FIELD_BIT_LENGTHS } from './constants';
+import {
+  ORACLE_PRICE_DECIMALS,
+  ORACLE_PRICE_FIELD_BIT_LENGTHS,
+} from './constants';
 import { getPedersenHash } from './hashes';
 import { StarkSignable } from './stark-signable';
 
 /**
  * Wrapper object to hash, sign, and verify an oracle price.
  */
-export class SignableOraclePrice extends StarkSignable<OraclePriceWithAssetId> {
+export class SignableOraclePrice extends StarkSignable<StarkwareOraclePrice> {
 
-  static fromPrice = SignableOraclePrice.fromPriceWithAssetName; // Alias.
+  static fromPriceWithMarket(
+    params: OraclePriceWithMarket,
+  ): SignableOraclePrice {
+    if (typeof params.market !== 'string') {
+      throw new Error('SignableOraclePrice.fromPrice: market must be a string');
+    }
+    const assetName = getSignedAssetName(params.market);
+    return SignableOraclePrice.fromPriceWithAssetName({
+      ...params,
+      assetName,
+    });
+  }
 
   static fromPriceWithAssetName(
     params: OraclePriceWithAssetName,
   ): SignableOraclePrice {
-    const assetNameBn = utf8ToBn(params.assetName, ORACLE_PRICE_FIELD_BIT_LENGTHS.assetName);
-    const oracleNameBn = utf8ToBn(params.oracleName, ORACLE_PRICE_FIELD_BIT_LENGTHS.oracleName);
+    if (typeof params.assetName !== 'string') {
+      throw new Error('SignableOraclePrice.fromPrice: assetName must be a string');
+    }
+    if (typeof params.oracleName !== 'string') {
+      throw new Error('SignableOraclePrice.fromPrice: oracleName must be a string');
+    }
+    if (typeof params.humanPrice !== 'string') {
+      throw new Error('SignableOraclePrice.fromPrice: humanPrice must be a string');
+    }
+    if (typeof params.isoTimestamp !== 'string') {
+      throw new Error('SignableOraclePrice.fromPrice: isoTimestamp must be a string');
+    }
+
+    const signedAssetId = getSignedAssetId(params.assetName, params.oracleName);
+
+    const signedPrice = new Big(params.humanPrice);
+    signedPrice.e += ORACLE_PRICE_DECIMALS;
 
-    const signedAssetId = assetNameBn
-      .iushln(ORACLE_PRICE_FIELD_BIT_LENGTHS.oracleName)
-      .iadd(oracleNameBn);
+    const expirationEpochSeconds = isoTimestampToEpochSeconds(params.isoTimestamp);
 
     return new SignableOraclePrice({
-      signedAssetId: signedAssetId.toString(16),
-      price: params.price,
-      isoTimestamp: params.isoTimestamp,
+      signedAssetId,
+      signedPrice: signedPrice.round(RoundingMode.RoundHalfEven).toFixed(0),
```

Comment:
nit: why round instead of expecting it to already be an integer?

Reply:
I was on the fence before, but think you're right it should throw instead

Reply:
I was on the fence before, but think you're right it should throw instead